docs/2019-12-16-Question-Mark-Variables-With-Ensures.md

```diff
@@ -0,0 +1,136 @@
+This design document describes four typical example language constructs or mathematical functions that have unspecified or nondeterministic behaviors and shows their axiomatization in Kore. We also propose the corresponding K frontend syntax to define some of these behaviors. The frontend syntax uses question mark variables `?X`, which when translated to Kore axioms become existentially quantified, and use `ensures` to specify the logical constraints that `?X` should satisfy. The `ensures` clause will be translated to the RHS in Kore axioms. 
```

Comment:
Below I've added a comment referring to something in the middle of one of these long lines, and I guess it's harder to resolve the reference than it could be. Why not 80-chars lines?